lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1112,6 +1116,13 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
             # TODO: remove this IF condition once LP#2076599 is fixed in Juju.
             return
 
+        # update any orchestrators about planned units this is useful when a new data node joins
+        # the orchestrator needs to know the planned units to determine if it can start or not
```

Comment:
in which case, shouldn't this only happen _after_ the unit starts? meaning at the end of the start process?